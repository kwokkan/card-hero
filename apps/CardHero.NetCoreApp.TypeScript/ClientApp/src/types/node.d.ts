declare const module: {
    exports: {}
};

declare const process: {
    env: {
        NODE_ENV: "production" | "development"
    }
};
