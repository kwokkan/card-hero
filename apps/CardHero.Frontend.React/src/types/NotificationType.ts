export enum NotificationType {
    Default,

    Info,

    Success,

    Warning,

    Danger,
}
