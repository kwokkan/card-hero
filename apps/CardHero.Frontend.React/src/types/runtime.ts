export interface Action {
    (): void;
}

export interface Func<TResult> {
    (): TResult;
}
