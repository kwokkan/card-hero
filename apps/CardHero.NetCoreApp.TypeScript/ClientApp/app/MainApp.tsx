import React, { Component, Fragment } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { IUserModel } from "../clients/clients";
import { Card } from "../components/card/Card";
import { CardApp } from "../components/card/CardApp";
import { CollectionApp } from "../components/collection/CollectionApp";
import { Deck } from "../components/deck/Deck";
import { DeckApp } from "../components/deck/DeckApp";
import { Game } from "../components/game/Game";
import { GameApp } from "../components/game/GameApp";
import { HomeApp } from "../components/home/HomeApp";
import { AppBootstrap } from "../components/shared/AppBootstrap";
import { ErrorBoundary } from "../components/shared/ErrorBoundary";
import { NavMenu } from "../components/shared/NavMenu";
import { StoreApp } from "../components/store/StoreApp";
import { AccountContext } from "../contexts/AccountContext";
import { AccountService } from "../services/AccountService";

interface IMainAppState {
    user?: IUserModel;
    setUser: (user: IUserModel) => void;
}

export class MainApp extends Component<{}, IMainAppState> {
    static contextType = AccountContext;

    constructor(props) {
        super(props);

        this.state = {
            setUser: (user) => {
                this.setState({
                    user: user
                })
            }
        };
    }

    async componentDidMount() {
        const user = await AccountService.getAccount();

        if (user) {
            this.context.setUser(user);
            this.setState({
                user: user
            });
        }
    }

    render() {
        return (
            <ErrorBoundary>
                <AccountContext.Provider value={this.state}>
                    <BrowserRouter basename={AppBootstrap.url()}>
                        <NavMenu appName={Constants.AppName} user={this.state.user} />

                        <div className="container-fluid body-content">
                            <div className="row">
                                <Switch>
                                    <Route exact path="/">
                                        <HomeApp />
                                    </Route>

                                    <Route path="/Card"
                                        render={({ match: { path } }) => (
                                            <Fragment>
                                                <Route exact path={`${path}/`}>
                                                    <CardApp routePrefix={path} />
                                                </Route>

                                                <Route path={`${path}/:id`}
                                                    render={({ match: { params, url } }) => (
                                                        <Route exact path={url}>
                                                            <Card id={params.id as number} />
                                                        </Route>
                                                    )}>
                                                </Route>
                                            </Fragment>
                                        )}>
                                    </Route>


                                    <Route path="/Game"
                                        render={({ match: { path } }) => (
                                            <Fragment>
                                                <Route exact path={`${path}/`}>
                                                    <GameApp routePrefix={path} />
                                                </Route>

                                                <Route path={`${path}/:id`}
                                                    render={({ match: { params, url } }) => (
                                                        <Route exact path={url}>
                                                            <Game id={params.id as number} />
                                                        </Route>
                                                    )}>
                                                </Route>
                                            </Fragment>
                                        )}>
                                    </Route>

                                    <Route path="/Store">
                                        <StoreApp />
                                    </Route>

                                    <Route path="/Collection">
                                        <CollectionApp />
                                    </Route>

                                    <Route path="/Deck"
                                        render={({ match: { path } }) => (
                                            <Fragment>
                                                <Route exact path={`${path}/`}>
                                                    <DeckApp routePrefix={path} />
                                                </Route>

                                                <Route path={`${path}/:id`}
                                                    render={({ match: { params, url } }) => (
                                                        <Route exact path={url}>
                                                            <Deck id={params.id as number} />
                                                        </Route>
                                                    )}>
                                                </Route>
                                            </Fragment>
                                        )}>
                                    </Route>
                                </Switch>
                            </div>
                        </div>
                    </BrowserRouter>
                </AccountContext.Provider>
            </ErrorBoundary>
        );
    }
}
