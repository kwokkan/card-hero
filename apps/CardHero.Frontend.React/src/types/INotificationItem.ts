import { NotificationType } from "./NotificationType";

export interface INotificationItem {
    title?: string;
    message: string | JSX.Element;
    type?: NotificationType;
}
