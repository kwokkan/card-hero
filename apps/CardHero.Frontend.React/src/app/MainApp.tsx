import React, { useState } from "react";
import { IUserModel } from "../clients/clients";
import { ErrorBoundary } from "../components/shared/ErrorBoundary";
import { IAccountContextProps } from "../contexts/AccountContext";
import { AccountContextProvider } from "../contexts/AccountContextProvider";
import { INotificationContextProviderProps, NotificationContextProvider } from "../contexts/NotificationContextProvider";
import { INotificationItem } from "../types/INotificationItem";
import { getRoutePrefix } from "../utils/route";
import { Main } from "./Main";

interface IMainAppProps {
    appName: string;
    baseUrl?: string;
    defaultNotifications?: INotificationItem[];
}

export function MainApp(props: IMainAppProps) {
    const [userState, setUserState] = useState<IUserModel>();

    const contextProps: IAccountContextProps = {
        user: userState,
        setUser: setUserState
    };

    const notificationContextProps: INotificationContextProviderProps = {
        notifications: props.defaultNotifications
    };

    const baseUrl = getRoutePrefix(props.baseUrl);

    return (
        <ErrorBoundary>
            <AccountContextProvider value={contextProps}>
                <NotificationContextProvider value={notificationContextProps}>
                    <Main appName={props.appName} baseUrl={baseUrl} />
                </NotificationContextProvider>
            </AccountContextProvider>
        </ErrorBoundary>
    );
}
