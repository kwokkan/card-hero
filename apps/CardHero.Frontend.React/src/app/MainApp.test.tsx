import React from "react";
import renderer from "react-test-renderer";
import { INotificationItem } from "../types/INotificationItem";
import { NotificationType } from "../types/NotificationType";
import { MainApp } from "./MainApp";

test("<MainApp /> renders correctly without notifications", () => {
    const tree = renderer
        .create(
            <MainApp appName="Test app" />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

test("<MainApp /> renders correctly with notifications", () => {
    const notifications: INotificationItem[] = [
        {
            message: "My info message",
            title: "My info title",
            type: NotificationType.Info
        },
        {
            message: "My danger message",
            title: "My danger message",
            type: NotificationType.Danger
        }
    ];
    const tree = renderer
        .create(
            <MainApp appName="Test app" defaultNotifications={notifications} />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
