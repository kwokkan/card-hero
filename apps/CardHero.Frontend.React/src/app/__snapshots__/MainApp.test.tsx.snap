// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainApp /> renders correctly with notifications 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark navbar-expand-md fixed-top"
  >
    <button
      aria-controls="navbar-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbar-header"
      data-toggle="collapse"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-fw "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Test app"
        className="d-inline-block align-top"
        height="32"
        src="/favicon.ico"
        width="32"
      />
       
      Test app
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbar-header"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Card"
            onClick={[Function]}
          >
            Cards
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Game"
            onClick={[Function]}
          >
            Games
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Store"
            onClick={[Function]}
          >
            Store
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://github.com/kwokkan/card-hero"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-github fa-fw "
              data-icon="github"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Fork me
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/swagger"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-code fa-fw "
              data-icon="code"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M414.8 40.79L286.8 488.8C281.9 505.8 264.2 515.6 247.2 510.8C230.2 505.9 220.4 488.2 225.2 471.2L353.2 23.21C358.1 6.216 375.8-3.624 392.8 1.232C409.8 6.087 419.6 23.8 414.8 40.79H414.8zM518.6 121.4L630.6 233.4C643.1 245.9 643.1 266.1 630.6 278.6L518.6 390.6C506.1 403.1 485.9 403.1 473.4 390.6C460.9 378.1 460.9 357.9 473.4 345.4L562.7 256L473.4 166.6C460.9 154.1 460.9 133.9 473.4 121.4C485.9 108.9 506.1 108.9 518.6 121.4V121.4zM166.6 166.6L77.25 256L166.6 345.4C179.1 357.9 179.1 378.1 166.6 390.6C154.1 403.1 133.9 403.1 121.4 390.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4L121.4 121.4C133.9 108.9 154.1 108.9 166.6 121.4C179.1 133.9 179.1 154.1 166.6 166.6V166.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            API
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/SignIn"
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid body-content"
  >
    <div
      className="notifications mr-3"
    >
      <div
        className="notification-item toast fade show"
      >
        <div
          className="toast-header"
        >
          <span
            className="badge badge-pill badge-secondary mr-2 badge-info"
          >
             
          </span>
          <strong
            className="mr-auto"
          >
            My info title
          </strong>
          <button
            aria-label="Close"
            className="ml-2 mb-1 close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          className="toast-body"
        >
          My info message
        </p>
      </div>
      <div
        className="notification-item toast fade show"
      >
        <div
          className="toast-header"
        >
          <span
            className="badge badge-pill badge-secondary mr-2 badge-danger"
          >
             
          </span>
          <strong
            className="mr-auto"
          >
            My danger message
          </strong>
          <button
            aria-label="Close"
            className="ml-2 mb-1 close"
            onClick={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
        </div>
        <p
          className="toast-body"
        >
          My danger message
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="jumbotron text-center"
        >
          <h1
            className="display-3"
          >
            Test app
          </h1>
          <p
            className="lead"
          >
            Test app
             is a dynamic action game.
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Cards
                </h4>
                <p
                  className="card-text"
                >
                  Discover over 1000 cards.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Card"
                  onClick={[Function]}
                >
                  View Cards
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Decks
                </h4>
                <p
                  className="card-text"
                >
                  Create you own decks.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Deck"
                  onClick={[Function]}
                >
                  View Decks
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Battle
                </h4>
                <p
                  className="card-text"
                >
                  Battle against other players.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Game"
                  onClick={[Function]}
                >
                  View Games
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`<MainApp /> renders correctly without notifications 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark navbar-expand-md fixed-top"
  >
    <button
      aria-controls="navbar-header"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbar-header"
      data-toggle="collapse"
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-fw "
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 96C0 78.33 14.33 64 32 64H416C433.7 64 448 78.33 448 96C448 113.7 433.7 128 416 128H32C14.33 128 0 113.7 0 96zM0 256C0 238.3 14.33 224 32 224H416C433.7 224 448 238.3 448 256C448 273.7 433.7 288 416 288H32C14.33 288 0 273.7 0 256zM416 448H32C14.33 448 0 433.7 0 416C0 398.3 14.33 384 32 384H416C433.7 384 448 398.3 448 416C448 433.7 433.7 448 416 448z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Test app"
        className="d-inline-block align-top"
        height="32"
        src="/favicon.ico"
        width="32"
      />
       
      Test app
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbar-header"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Card"
            onClick={[Function]}
          >
            Cards
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Game"
            onClick={[Function]}
          >
            Games
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Store"
            onClick={[Function]}
          >
            Store
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="https://github.com/kwokkan/card-hero"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-github fa-fw "
              data-icon="github"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            Fork me
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/swagger"
            rel="noreferrer noopener"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-code fa-fw "
              data-icon="code"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M414.8 40.79L286.8 488.8C281.9 505.8 264.2 515.6 247.2 510.8C230.2 505.9 220.4 488.2 225.2 471.2L353.2 23.21C358.1 6.216 375.8-3.624 392.8 1.232C409.8 6.087 419.6 23.8 414.8 40.79H414.8zM518.6 121.4L630.6 233.4C643.1 245.9 643.1 266.1 630.6 278.6L518.6 390.6C506.1 403.1 485.9 403.1 473.4 390.6C460.9 378.1 460.9 357.9 473.4 345.4L562.7 256L473.4 166.6C460.9 154.1 460.9 133.9 473.4 121.4C485.9 108.9 506.1 108.9 518.6 121.4V121.4zM166.6 166.6L77.25 256L166.6 345.4C179.1 357.9 179.1 378.1 166.6 390.6C154.1 403.1 133.9 403.1 121.4 390.6L9.372 278.6C-3.124 266.1-3.124 245.9 9.372 233.4L121.4 121.4C133.9 108.9 154.1 108.9 166.6 121.4C179.1 133.9 179.1 154.1 166.6 166.6V166.6z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
             
            API
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/SignIn"
          >
            Sign in
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid body-content"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <div
          className="jumbotron text-center"
        >
          <h1
            className="display-3"
          >
            Test app
          </h1>
          <p
            className="lead"
          >
            Test app
             is a dynamic action game.
          </p>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Cards
                </h4>
                <p
                  className="card-text"
                >
                  Discover over 1000 cards.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Card"
                  onClick={[Function]}
                >
                  View Cards
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Decks
                </h4>
                <p
                  className="card-text"
                >
                  Create you own decks.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Deck"
                  onClick={[Function]}
                >
                  View Decks
                </a>
              </div>
            </div>
          </div>
          <div
            className="col-lg-4"
          >
            <div
              className="card text-center"
            >
              <div
                className="card-body"
              >
                <h4
                  className="card-title"
                >
                  Battle
                </h4>
                <p
                  className="card-text"
                >
                  Battle against other players.
                </p>
                <a
                  className="btn btn-primary"
                  href="/Game"
                  onClick={[Function]}
                >
                  View Games
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
